user_assigned_identity/variables.tf

```diff
@@ -22,3 +22,8 @@ variable "role_assignments" {
   description = "Application Roles to assign the Managed Identity"
   type        = list(string)
 }
+
+variable "group_membership" {
```

Comment:
Add type constraint `type = list(string)`

Reply:
How about setting it to set? There should not be any duplicated object_ids passed to the variable